examples/sk_linear_advection.py

```diff
@@ -0,0 +1,158 @@
+from dcore import *
+from firedrake import Expression, \
+    VectorFunctionSpace, PeriodicIntervalMesh, ExtrudedMesh
+from firedrake import exp, sin, ds_b
+import numpy as np
+
+nlayers = 50  # horizontal layers
+columns = 50  # number of columns
+L = 3.0e5
+m = PeriodicIntervalMesh(columns, L)
+
+# build volume mesh
+H = 1.0e4  # Height position of the model top
+mesh = ExtrudedMesh(m, layers=nlayers, layer_height=H/nlayers)
+
+# Space for initialising velocity
+W_VectorCG1 = VectorFunctionSpace(mesh, "CG", 1)
+W_CG1 = FunctionSpace(mesh, "CG", 1)
+
+# vertical coordinate and normal
+z = Function(W_CG1).interpolate(Expression("x[1]"))
+k = Function(W_VectorCG1).interpolate(Expression(("0.","1.")))
+
+fieldlist = ['u', 'rho', 'theta']
+timestepping = TimesteppingParameters(dt=10.0)
+output = OutputParameters(dirname='sk_linear', dumplist=['u'])
+parameters = CompressibleParameters()
+
+state = CompressibleState(mesh, vertical_degree=1, horizontal_degree=1,
+                          family="CG",
+                          z=z, k=k,
+                          timestepping=timestepping,
+                          output=output,
+                          parameters=parameters,
+                          fieldlist=fieldlist)
+
+# Initial conditions
+u0, theta0, rho0 = Function(state.V[0]), Function(state.V[2]), Function(state.V[1])
+
+# Thermodynamic constants required for setting initial conditions
+# and reference profiles
+g = parameters.g
+N = parameters.N
+p_0 = parameters.p_0
+c_p = parameters.cp
+R_d = parameters.R_d
+kappa = parameters.kappa
+
+# N^2 = (g/theta)dtheta/dz => dtheta/dz = theta N^2g => theta=theta_0exp(N^2gz)
+Tsurf = 300.
+thetab = Tsurf*exp(N**2*z/g)
+
+theta_b = Function(state.V[2]).interpolate(thetab)
+rho_b = Function(state.V[1])
+
+# Calculate hydrostatic Pi
```

Comment:
I think this hydrostatic initialisation should be moved to an initialise method in the CompressibleState class, I only hesitate because I'm not sure if it works for anything other than the vertical slice setup.